### PR TITLE
WT-3754 serial updates to the oldest to consider pinned ts for visibility

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@agorrod please review

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -339,9 +342,23 @@ __wt_update_obsolete_check(
...
-		    S2C(session)->txn_global.last_running;
+	if (count > 20 && page->modify != NULL) {
+		page->modify->obsolete_check_txn = txn_global->last_running;
+#ifdef HAVE_TIMESTAMPS
+		if (txn_global->has_pinned_timestamp) {
+			WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
+			    __wt_timestamp_set(
+				&page->modify->obsolete_check_timestamp,
+				&txn_global->pinned_timestamp));
+			page->modify->has_obsolete_check_timestamp = true;
```
Comment (michaelcahill) [inline src/btree/row_modify.c]: @sulabhM, can you explain why we need this boolean?

Without it, `page->modify->obsolete_check_timestamp` should be zero, which means it should always pass the `__wt_txn_visible_all` check below.

File: src/include/serial.i [hunk 2]
```diff
@@ -309,11 +310,20 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_PAGE *page,
...
 	 */
 	if ((txn = page->modify->obsolete_check_txn) != WT_TXN_NONE) {
-		if (!__wt_txn_visible_all(session, txn, NULL)) {
+#ifdef HAVE_TIMESTAMPS
+		obsolete_timestamp =
+		    page->modify->has_obsolete_check_timestamp ?
+		    &page->modify->obsolete_check_timestamp : NULL;
+#else
+		WT_UNUSED(obsolete_timestamp);
+#endif
```
Comment (michaelcahill) [inline src/include/serial.i]: I think this block can be simplified to `obsolete_timestamp = WT_TIMESTAMP_NULL(&page->modify->obsolete_check_timestamp);`, then pass it unconditionally below.

**Comment by michaelcahill** [comment]
Thanks @sulabhM, lgtm.  I'm going to kick off a patch build to see what impact this has on MongoDB performance before merging.

**Comment by agorrod** [comment]
lgtm - I'm going to merge

**Pull request merged**